### ISSUE TITLE
Design of Host module

### ISSUE THREAD

**Issue by LeonYuAng3NT** [SmileWuji/tea-party-deception#3]
- I really like and support the idea of super host module.However, I feel that the convention shouldn't be implemented in that format because it's confusing and hard to follow. I mean why not just create an internal API inside the Host module and make sure it's not able to exposed to the user? 
This part is great: ”the server will put all the commands into a JSON and throw them to the Ui upon each state transition.“ but it leads to my previous question on why not API with Regex or optional params?  

- Unless I'm mistaken, tpdhost.Announcer is essentially pipeline between the Ui and the host which makes me wonder why you want to make the Host module to be abstract? Why do we have direct communication between the Ui and the host ? Is the host the back-end logic? If so , why do we have it contain abstract functions?

**Comment by SmileWuji** [comment]
You are right my head was kicked by a donkey when I wrote the host.

The issue #1 will be working on this issue as well. For data transmission, it will be entirely encapsulated by the server modules. 

Further decision on how the server should behave (i.e. how the data get transmitted) is required for this issue.

**Comment by LeonYuAng3NT** [comment]
This is now a stretch goal, however, a really good to have feature for the overall game.  Game engine covers the server and host for prototype version.

**Comment by SmileWuji** [comment]
Clarified with Leon in person yesterday. The responsibility of the host is to manage the logic (of this game) + Local Area Network multiplayer connection.
